include/stdio.h

```diff
@@ -225,6 +216,49 @@ FILE *popen(FAR const char *command, FAR const char *mode);
 int pclose(FILE *stream);
 #endif
 
+/* These APIs are not implemented and/or can be synthesized from
+ * supported APIs.
+ */
+
+#ifdef CONFIG_HAVE_INLINE
+static inline int putc(int c, FAR FILE *s)
```

Comment:
why don't you make them real (non-inline) functions unconditionally?

Reply:
Because some compiler generate the bloat code for inline function:
https://github.com/apache/incubator-nuttx/commit/3b53cd1e578de1e6bddfae605452a77323e5a70b

Reply:
i'm suggesting to make them non-inline functions in *.c.

Reply:
But these functions are very simple, non-inline increase the code size and decrease the performance. And the same usage spread many place.

Reply:
i can understand the performance argument for ctype functions.
but i'm not sure about these FILE* functions. have you measured?

if you really care the (IMO marginal) performance gain,
i think the traditional convention is to have a macro version overriding the non-inline function
so that any app which is not happy with the macro version can do "#undef getc" before using it.
like:

    int getc(FAR FILE *stream);
    #define getc(s) fgetc(s)

Reply:
> i can understand the performance argument for ctype functions.
> but i'm not sure about these FILE* functions. have you measured?
> 

I mean the ctype like functions when I mention the performance. If we have to use #ifdef/#endif for ctype function, why we use the different aproach for other functions?  I prefer to keep the consistency.

> if you really care the (IMO marginal) performance gain,
> i think the traditional convention is to have a macro version overriding the non-inline function
> so that any app which is not happy with the macro version can do "#undef getc" before using it.
> like:
> 
> ```
> int getc(FAR FILE *stream);
> #define getc(s) fgetc(s)
> ```

But I can't use this approach(add #undef ...), because the code is in the 3rd party library, I don't want to change their code if the code totally confirm the standard.

Reply:
i prefer to have less #ifdef CONFIG_xxx.
my suggestion (not to use inline or macro) is consistent with the other functions.
inlined things are minority.

i can argue the 3rd part library is not totally standard-conforming as standards allow these functions to be implemented as macros.
they even suggest the #undef tricks. see APPLICATION USAGE in
https://pubs.opengroup.org/onlinepubs/9699919799/functions/getc.html

Reply:
I can change all inline functions with #ifdef CONFIG_HAVE_INLINE to normal function if no other people disagree.

Reply:
@yamt done, please take a look.